contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -128,40 +112,43 @@ contract AxelarAmplifierGateway is Ownable, IAxelarAmplifierGateway {
         for (uint256 i; i < commandsLength; ++i) {
             bytes32 commandId = commandIds[i];
 
-            if (isCommandExecuted(commandId)) continue; /* Ignore if duplicate commandId received */
+            // Ignore if commandId is already executed
+            if (isCommandExecuted(commandId)) {
+                continue;
+            }
 
-            bytes4 commandSelector;
             bytes32 commandHash = keccak256(abi.encodePacked(commands[i]));
 
             if (commandHash == SELECTOR_APPROVE_CONTRACT_CALL) {
-                commandSelector = AxelarAmplifierGateway.approveContractCall.selector;
+                _approveContractCall(slot, params[i], commandId);
             } else if (commandHash == SELECTOR_TRANSFER_OPERATORSHIP) {
-                if (!allowOperatorshipTransfer) continue;
+                if (!allowOperatorshipTransfer) {
+                    continue;
+                }
 
                 allowOperatorshipTransfer = false;
-                commandSelector = AxelarAmplifierGateway.transferOperatorship.selector;
+
+                _transferOperatorship(params[i]);
             } else {
-                continue; /* Ignore if unknown command received */
+                revert InvalidCommand(commandHash);
             }
 
-            // Prevent a re-entrancy from executing this command before it can be marked as successful.
             slot.commands[commandId] = true;
-            (bool success, ) = address(this).call(abi.encodeWithSelector(commandSelector, params[i], commandId));
 
-            if (success) {
-                // slither-disable-next-line reentrancy-events
-                emit Executed(commandId);
-            } else {
-                slot.commands[commandId] = false;
-            }
+            // slither-disable-next-line reentrancy-events
+            emit Executed(commandId);
         }
     }
 
-    /******************\
-    |* Self Functions *|
-    \******************/
+    /**********************\
+    |* Internal Functions *|
+    \**********************/
 
-    function approveContractCall(bytes calldata params, bytes32 commandId) external onlySelf {
+    function _approveContractCall(
+        AxelarAmplifierGatewayStorage storage slot,
```

Comment:
Maybe there is no need to pass the storage slot as it can be accessed in the function by `_axelarAmplifierGatewayStorage()`

Reply:
confirmed it's cheaper to retrieve it directly instead of creating a variable